content/docs/user-guide/testing/best-practices/_index.md

```diff
@@ -0,0 +1,327 @@
+
+A test which must target platform-specific behavior should conditionally disable itself when that dependency is unavailable. Tests should *AVOID* performing conditional steps based on their environment, as this makes "the same test" significantly more difficult to investigate when it automatically behaves differently on different machines.
+
+### Negative Assertions
+
+Tests should positively verify an expected result occurred, instead of trying to negatively verify that "no bad side-effects" happened. This creates a problem because, unlike the correct result, the set of all potential incorrect results is effectively infinite. Consider the following code:
+
+```cpp
+int Math::FastSquare( const int number )
+{
+    return number + number; // bug!
+}
+```
+
+```cpp
+TEST(MathTests, FastSquare_Integer_Squared)
+{
+    //act
+    int result = Math::FastSquare(10);
+    //assert
+    EXPECT_NE(result), 0);
+    EXPECT_NE(result, -1);
+    EXPECT_NE(result, 10);
+}
+```
+
+The example above contains a bug where the code under test does not actually compute the square of a number. The test correctly verifies the computation is not equal to a few values known to not be the square of 10, however it never attempts to identify the correct answer. Due to this, the test cannot catch the bug!
+
+{{< note >}}
+Negative assertions are different from negative test cases. A negative test case should set up an error-scenario, and then positively identify that one correct error-response is emitted. Negative cases should also avoid negative assertions.
+{{< /note >}}
+
+It can be okay to include negative assertions when there is also at least one positive assertion, but they should not be solely relied on. Focus on identifying the expected output.
+
+## Remember
+
+### Insanity Checks
+
+Make sure a new test is actually configured to run and report failures by temporarily editing the production code. Intentionally break the code in one way the test should detect, and run the test suite. If no failure occurs, investigate why!
+
+### Floating Point Assertions
+
+Floating-point numbers have problems with exact equality, as two floats can represent the "same" value but with slightly different rounding errors. Test frameworks provide special assertions that already account for many common issues:
+
+* C++ distinguishes between floats and doubles since they are assessed differently `EXPECT_FLOAT_EQ(val1, val2)` and `EXPECT_DOUBLE_EQ(val1, val2)`
+  * Custom error tolerance can also be provided to `EXPECT_NEAR(val1, val2, abs_error)`.
+* Python only has one floating point primitive, which is easily checked with `assert val1 == pytest.approx(val2)` or `unittest.assertAlmostEqual(val1, val2)`
```

Comment:
in Editor Python Bindings python (in editor tests) you can also use azlmb.math.Math_IsClose(val1, val2) an optional 3rd parameter for non-default tolerance can also be utilized if variance is larger than default.
`from azlmbr.math import Math_IsClose`